crates/ibc/src/core/context.rs

```diff
@@ -272,13 +285,27 @@ pub trait ExecutionContext: ValidationContext {
         client_state: Box<dyn ClientState>,
     ) -> Result<(), ContextError>;
 
+    /// Sets the upgraded `ClientState` in the upgrade path
+    fn store_upgraded_client_state(
```

Comment:
All new upgrade-related methods on `ValidationContext` and `ExecutionContext` are never used. Are they necessary? ibc-go doesn't seem to have any special methods to store an "upgraded" client/consensus state vs a "normal" one

Reply:
`ValidationContext` and `ExecutionContext` are the contexts specific to handling all messages in `MsgEnvelope`. This is something else. We'd need some kind of [`HandleUpgradeProposal`](https://github.com/cosmos/ibc-go/blob/295bab01f633633e0d6136ba9b974c0e248edbe5/modules/core/02-client/keeper/proposal.go#L73) contexts (validation & execution), which chains can use to do upgrade proposals. However, I believe you had trouble creating such an abstraction?

Reply:
[736a7d2](https://github.com/cosmos/ibc-rs/pull/673/commits/736a7d2e770d46d49e23f3f708fa3340c247c0f7)